### PR TITLE
Add check for already existed subscriptions during resumption in VehicleInfoPlugin

### LINKED ISSUE #2283: SDL sends SubscribeVehicleData by resumption second app
### Bug Report
SDL sends SubscribeVehicleData by registration second app

##### Reproduction Steps
1. Register app1 and app2
2. App1 subscribe VD and App2 subscribe the same VD
3. Reconnect apps

##### Expected Behavior
1. SDL resumes SubscribeVehicleData for app1 and sends SubscribeVehicleData request to HMI
2. SDL resumes SubscribeVehicleData for app2 and does not send SubscribeVehicleData request to HMI

##### Observed Behavior
SDL sends SubscribeVehicleData requests to HMI during resumption of both apps.

##### OS & Version Information
* OS/Version: Ubuntu 16.04
* SDL Core Version: develop branch (2ec732c56e3f06b7043cf02cf98109720d6ff38a)
* Testing Against: ATF: [develop branch](https://github.com/smartdevicelink/sdl_atf/tree/develop)
Script: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1936 (004_Resumption_success_flow_2_apps.lua)

##### Attach
[SmartDeviceLinkCore_subscribeVD_resumption_2app.log.tar.gz](https://github.com/smartdevicelink/sdl_core/files/2118442/SmartDeviceLinkCore_subscribeVD_resumption_2app.log.tar.gz)

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #2283 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
During regression checks incorrect scripts were found, this fact was reflected in corresponding issue https://github.com/smartdevicelink/sdl_atf_test_scripts/issues/2385, and this problem is solved in pull request https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2386. PR with actualized scripts needs to be merged after merging this PR.

### Summary
This pull request is intended to address issue with incorrect behavior SDL during resumption, when SDL sent SubscribeVehicleData request to HMI repeatedly for already resumed subscriptions. Due to this fix some changes were added to methods of VehicleInfoPlugin class. This changes provide check for already resumed subscriptions (similar check is implemented in SubscribeVehicleDataRequest class, for example).

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by ychernysheva** [comment]
@theresalech This PR is ready for review. Please also look at related issue smartdevicelink/sdl_atf_test_scripts#2385 and PR with corresponding fix smartdevicelink/sdl_atf_test_scripts#2386.
Thank you!

**Comment by theresalech** [comment]
@ychernysheva Issue #2283 has not been raised by SDLC Members as priority for the Core 6.2 Release.  Please advise if Luxoft would like to prioritize this issue for Core 6.2.  Otherwise, we will work to include as time allows.

**Comment by okandul** [comment]
@theresalech Yes, we see it as a part of 6.2. Could you please prioritize it accordingly? Thanks in advance.

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc [hunk 1]
```diff
@@ -154,28 +154,50 @@ void VehicleInfoPlugin::UnsubscribeFromRemovedVDItems() {
...
   const auto& subscriptions = ext.Subscriptions();
+
   if (subscriptions.empty()) {
     LOG4CXX_DEBUG(logger_, "No vehicle data to subscribe. Exiting");
     return;
   }
 
-  for (const auto& item : subscriptions) {
-    msg_params[item] = true;
+  for (auto& ivi : subscriptions) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc:187]: move initial_length closer to this loop.

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc [hunk 2]
```diff
@@ -154,28 +154,50 @@ void VehicleInfoPlugin::UnsubscribeFromRemovedVDItems() {
...
   }
 
-  smart_objects::SmartObjectSPtr request =
-      application_manager::MessageHelper::CreateModuleInfoSO(
-          hmi_apis::FunctionID::VehicleInfo_SubscribeVehicleData,
-          *application_manager_);
-  (*request)[strings::msg_params] = msg_params;
-  application_manager_->GetRPCService().ManageHMICommand(request);
+  if (msg_params.length() > initial_length) {
+    smart_objects::SmartObjectSPtr request =
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc:194]: auto

Comment (ychernysheva) [reply src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc]: Done in commit 7571f74

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc (hunk 1 above)
Comment (ychernysheva) [reply src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc:188]: Done in commit 7571f74

**Pull request merged**